src/domain-services/flows/flow-search-service.ts

```diff
@@ -0,0 +1,652 @@
+      limit,
+      nextPageCursor,
+      prevPageCursor,
+      sortField,
+      sortOrder,
+      includeChildrenOfParkedFlows: shouldIncludeChildrenOfParkedFlows,
+      nestedFlowFilters,
+      status,
+    } = filters;
+
+    const orderBy: FlowOrderBy = this.buildOrderBy(sortField, sortOrder);
+
+    let { flowFilters } = filters;
+    const {
+      flowObjectFilters,
+      flowCategoryFilters,
+      pending: isPendingFlows,
+      commitment: isCommitmentFlows,
+      paid: isPaidFlows,
+      pledge: isPledgedFlows,
+      carryover: isCarryoverFlows,
+      parked: isParkedFlows,
+      pass_through: isPassThroughFlows,
+      standard: isStandardFlows,
+    } = filters;
+
+    // Returns an object like
+    // { name: 'Pending review',
+    // operation: 'IN'} []
+    const shortcutFilters: ShortcutCategoryFilter[] | null =
+      this.mapShortcutFilters(
+        isPendingFlows,
+        isCommitmentFlows,
+        isPaidFlows,
+        isPledgedFlows,
+        isCarryoverFlows,
+        isParkedFlows,
+        isPassThroughFlows,
+        isStandardFlows
+      );
+
+    // If shortcutFilter is defined
+    // We need to check if 'isPendingFlows' is true
+    // If so, we need to add the 'activeStatus' filter
+    // To 'false' by default if there is no status filter applied
+    if (
+      shortcutFilters &&
+      isPendingFlows &&
+      (!flowFilters || flowFilters.activeStatus === undefined)
+    ) {
+      flowFilters = flowFilters ?? {};
+      flowFilters.activeStatus = false;
+    }
+
+    // Once we've gathered all the filters, we need to determine the strategy
+    // to use in order to obtain the flowIDs
+    const strategy: FlowSearchStrategy = this.determineStrategy(
+      flowFilters,
+      flowObjectFilters,
+      flowCategoryFilters,
+      nestedFlowFilters,
+      shortcutFilters,
+      status,
+      orderBy
+    );
+
+    const offset = nextPageCursor ?? prevPageCursor ?? 0;
+
+    // We add 1 to the limit to check if there is a next page
+    const searchLimit = limit + 1;
+
+    const { flows, count } = await strategy.search({
+      models,
+      databaseConnection,
+      limit: searchLimit,
+      orderBy,
+      offset,
+      flowFilters,
+      flowObjectFilters,
+      flowCategoryFilters,
+      nestedFlowFilters,
+      // Shortcuts for categories
+      shortcutFilters,
+      statusFilter: status,
+      shouldIncludeChildrenOfParkedFlows,
+    });
+
+    // Remove the extra item used to check hasNextPage
+    const hasNextPage = flows.length > limit;
+    if (hasNextPage) {
+      flows.pop();
+    }
+
+    const flowIds: FlowId[] = [];
+    const flowWithVersion: Map<FlowId, number[]> = new Map<FlowId, number[]>();
+
+    // Obtain flow IDs and flow version IDs
+    for (const flow of flows) {
+      flowIds.push(flow.id);
+      if (!flowWithVersion.has(flow.id)) {
+        flowWithVersion.set(flow.id, []);
+      }
+      const flowVersionIDs = flowWithVersion.get(flow.id)!;
```

Comment:
Don't use non-null assertion operator! It should only be used in very rare circumstance where Typescript cannot infer the existence of a variable/method result. In this scenario, you could argue that `flowVersionIDs` can never be `undefined`, but you should have used [`getOrCreate`](https://github.com/UN-OCHA/hpc-api-core/blob/8a2a05fda25ac14af79ba5744f2f4a0cf6327765/src/util/index.ts#L205), which is a standardized method to do what you are doing here.

Since this isn't the only place where you're using non-null assertion operator, I will enable [`no-non-null-assertion` linting rule](https://typescript-eslint.io/rules/no-non-null-assertion/), to prevent such code from reaching code review in the future.

Reply:
Sure thing! I'm updating them all